mr/src/test/java/org/apache/iceberg/mr/hive/TestTables.java

```diff
@@ -83,12 +92,107 @@ public Tables tables() {
   public abstract String locationForCreateTableSQL(TableIdentifier identifier);
 
   /**
-   * If the {@link Catalogs#LOCATION} is needed for {@link Catalogs#loadTable(Configuration, Properties)} then this
-   * method should provide the location string. It should return <code>null</code> if the location is not needed.
-   * @param identifier The table identifier
-   * @return The location string for loadTable operation
+   * If creating the Hive table independently is needed for the given Catalog then this should return the Hive SQL
+   * string which is needed to be executed.
+   * @param identifier The table identifier (the namespace should be "default")
+   * @return The SQL string - which should be executed, null - if it is not needed.
+   */
+  public String createHiveTableSQL(TableIdentifier identifier) {
+    return String.format("CREATE TABLE default.%s STORED BY '%s' %s", identifier.name(),
```

Comment:
quick question: shouldn't we allow to pass in a custom database name via identifier.namespace()? any reason we have to force "default"?

Reply:
Added the possibility